activerecord/lib/active_record/associations/association_scope.rb

```diff
@@ -129,38 +113,57 @@ def next_chain_scope(scope, table, reflection, connection, assoc_klass, foreign_
         scope = scope.joins(join(foreign_table, constraint))
       end
 
-      def add_constraints(scope, owner, assoc_klass, refl, tracker)
-        chain = refl.chain
-        scope_chain = refl.scope_chain
-        connection = tracker.connection
+      class ReflectionProxy < SimpleDelegator # :nodoc:
+        attr_accessor :next
+        attr_reader :alias_name
+
+        def initialize(reflection, alias_name)
+          super(reflection)
+          @alias_name = alias_name
+        end
+
+        def all_includes; nil; end
+      end
 
-        tables = construct_tables(chain, assoc_klass, refl, tracker)
+      def get_chain(refl, association, tracker)
```

Comment:
Either the method signature or the block needs to change because if they're both `reflection` we'll get a shadowing outer local variable warning. It follows `#add_constraints` which has the same problem of needing the reflection variable twice. I'm trying to think of another name so this doesn't have to be abbreviated.